ui-v2/app/styles/components/action-group/skin.scss

```diff
@@ -20,7 +20,10 @@
 }
 %action-group input[type='radio']:checked + label:first-of-type,
 %action-group label:first-of-type:hover {
-  background-color: $gray;
+  background-color: $ui-gray-050;
```

Comment:
In Vault I [renamed the colors](https://github.com/hashicorp/vault/blob/b7655d78801a6484ec1407da278a410046ff1bbf/ui/app/styles/utils/_colors.scss) to friendlier names like you did with `$keyline-` below. How would you feel about using semantic names like that instead of the specific color levels?

Reply:
Hey @joshuaogle ,

Yeah good point, on one hand, me personally, I wouldn't want to use `-dark`, `-darker` like you did in vault as I think that will just lead to problems (`-darkerer`, `-not-as-dark-as-darker-but-darker-than-dark` etc 😄 ), and I think the `-100`, `-200`, `-300` solves that problem by being able to sensibly shim in temporary numbers whilst still making sense, and its clear enough that you can understand that it means a gradual color change from light to dark (the same as what dark, darker, darkest does, but without the possible problems when you need to specify an 'inbetween' color in the future)

For example if I'm a dev I know that if I need it a bit darker I just go up from 600 to 700, or from 600 to 800 if that's not quite right, I actually think we'll probably need more grays, in the past designers have always had way more than 10 across a project - grays are strange, 'boring' but probably the most used color 😄 

But on the other hand, for the `keyline-` thing, I definitely want to do more of these, but I want to be really careful in defining them, it seems like a simple enough task but it's one of those things that would be good to get a perfect balance between not having too many and not having enough. I'm still deliberating on the `keyline-` grouping itself, I'm pretty sure I'm going to replace it with a `%grey-box-100`, `%grey-box-200`, `%grey-box-300`, `%blue-box-100`, `%blue-box-500`, `%blue-box-900` which will be `background-color` + `border-color` + `color` (possibly `color`) extendible placeholders. Basically this:

https://github.com/johncowen/iris/commit/b6dcf8a00a91356e4fe72288c912a43a35ff8858#r29747574

..but I've taken it a bit further in my head at the minute, and figured it would be used for the alert boxes in your structure designs, and even buttons if we do it for all our ui colors (not just grey combinations). I might get change to drop something into iris today to illustrate a little more.  

Just to brain dump some more thinking here.. I figure backgrounds/borders/text-colors of boxes are generally very tightly coupled. For a grey box, the border will depend on what the background color is and in turn the text color will depend on what the background color is. I'm thinking these combinations can be tied up in these `%grey-box-100`, `%red-box-500` placeholders and just extended like so:

```css
.warning-alert {
  @extend %yellow-box-500;
  border-width: 1px;
}
/* this one only has a border on the bottom*/
.filter-bar {
  @extend %grey-box-500;
  border-bottom-width: 1px;
}
.delete-button {
  @extend %red-box-100;
  border-width: 1px;
}
.delete-button:hover {
  @extend %red-box-600;
  border-width: 1px;
}
```

...would essentially produce...

```css
.warning-alert {
  background-color: $ui-yellow-200;
  border-color: $ui-yellow-600;
  color: $ui-yellow-700;
  border-style: solid;
  border-width: 1px;
}
.filter-bar {
  background-color: $ui-yellow-200;
  border-color: $ui-yellow-600;
  color: $ui-yellow-700;
  border-style: solid;
  border-bottom-width: 1px;
}
.delete-button {
  background-color: $ui-white;
  border-color: $ui-red-500;
  color: $ui-red-500;
  border-style: solid;
  border-width: 1px;
}
.delete-button:hover {
  background-color: $ui-red-600;
  border-color: $ui-red-600;
  border-style: solid;
  border-width: 1px;
}
```

We'd have to carefully define these combo's in your Structure file of course, and even writing this out now I'm even thinking whether we'd ever need to distinguish between `%brand-grey-box` and `%ui-grey-box`? Pretty sure we don't but just to flag up here.

 Let me know what you think

Reply:
@johncowen I think I agree with all of your points here, summarized as the following guidelines: 

✅ Avoid relative naming (dark vs darker vs darkest)
✅ Semantic naming should be enough to be helpful but not so overdone that there are too many options to choose from.

I wonder if it would be helpful to keep a list of CSS guidelines in a readme (or somewhere else) so that all our contributors (including myself) can know our best practices and the reasoning behind them when we add or edit Consul CSS. 

In terms of this specific case, I think `box` might be too broad of a term. It could be a great opportunity to create semantic names for button colors (maybe it wasn't obvious but this element is actually a button component in Structure), like this:

```
--color-button-primary: var(--ui-blue-500);
--color-button-secondary: var(--ui-blue-050);
--color-button-ghost: var(--ui-gray-100);
--color-button-warning: var(--ui-color-danger);
```

Then, we can do the same for hover or active states, OR... we could make some adjustments such that we can apply a lighten or darken method that could be applied to all buttons (the latter option might require a bit of a conversation with the designers, and the former option can be implemented at least for now). 

cc @joshuaogle

Reply:
@johncowen Yep, I totally understand the concern around relative named variables. I think I'd argue that if you do have `-not-as-dark-as-darker-but-darker-than-dark`, well then you have too many grays and already screwed up 😄 

Ultimately though, you're right that those aren't semantic anyway. I think we'll still want to override Bulma's variables like `$gray-lighter`, but for ours we should have better names and I think @opihana is on the right track with her suggestions :+1:

Reply:
Also, since I was merely changing this variable from `$gray` to `$ui-gray-050`, we could tackle the semantic button naming in another PR

Reply:
@opihana, personally I'd prefer to have a `%button-warning` to extend that does the padding/layout/decoration and color all in one shot:

```css
button.type-delete {
  @extend %button-warning;
}
```
That would set the color/padding/layout/decoration plus all the `:hover`, `:focus` states all in one shot.

Once you have this, having variables for:

```css
--color-button-warning: var(--ui-color-danger);
// multiplied by all the buttons, multiplied by all the states
```

..just seems like overkill/lots of code that you'd never use when you have a `%button-warning` to extend.

Also you may need other things that aren't buttons having a `warning` color, then you get people writing things like:

```css
.alert-notification {
  border-color: var(--color-button-warning);
}
```
or

```css
--color-button-warning: var(--ui-color-danger);
--color-notification-warning: var(--ui-color-danger);
--color-something-else-warning: var(--ui-color-danger);
// multiplied by lots
```

Re: `box`

It needs to be a fairly generic word as I can use it on anything that needs a colored background/border (buttons, notifications, alerts, sectioned areas in the page, menus, allsorts). Its about background-colors, border-colors and text colors being fairly tightly coupled from a design perspective - you want to make sure that those colors are contrast-y enough, so you can go some way in grouping those combinations into `box`'es 

Happy for another word, but it should 'mean' 'thing with possibly a background, possibly a border that can possibly contain text'. `shape`, `block`, `swatch`? `container` I don't like but thought I'd add here anyway. `swatch` I almost like as its color-y but I think it means something else.

Dunno lets chat later, maybe we could hook up with @joshuaogle and nut it out.